pkg/vault/contracts/BatchRouterCommon.sol

```diff
@@ -117,13 +117,17 @@ abstract contract BatchRouterCommon is RouterCommon {
             // Erases delta, in case more than one batch router operation is called in the same transaction.
             _currentSwapTokenInAmounts().tSet(tokenIn, 0);
             _currentSwapTokensIn().remove(tokenIn);
+            _settledTokenAmounts().tSet(tokenIn, 0);
```

Comment:
This code is shared by the batch router and the CLR.
This is done at a different level for the batch router though, so if we set it to 0 here unconditionally we'll be setting it twice.

I'm not entirely against that, but maybe we should zero them out after using it (and `_settledTokenAmounts` is not used in this function).

Reply:
Ok; trying to set directly in the CLR. Will be more code, but should only affect the CLR.